src/objective-c/tests/ThirdPartyTests/Libuv/LibuvTimerTests.m

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#import <XCTest/XCTest.h>
+
+#import "test/runner.h"
+#import "test/task.h"
+
+TEST_DECLARE(timer)
+TEST_DECLARE(timer_init)
+TEST_DECLARE(timer_again)
+TEST_DECLARE(timer_start_twice)
+TEST_DECLARE(timer_order)
+TEST_DECLARE(timer_huge_timeout)
+TEST_DECLARE(timer_huge_repeat)
+TEST_DECLARE(timer_run_once)
+TEST_DECLARE(timer_from_check)
+TEST_DECLARE(timer_is_closing)
+TEST_DECLARE(timer_null_callback)
+TEST_DECLARE(timer_early_check)
+
+TASK_LIST_START
+TEST_ENTRY(timer)
+TEST_ENTRY(timer_init)
+TEST_ENTRY(timer_again)
+TEST_ENTRY(timer_start_twice)
+TEST_ENTRY(timer_order)
+TEST_ENTRY(timer_huge_timeout)
+TEST_ENTRY(timer_huge_repeat)
+TEST_ENTRY(timer_run_once)
+TEST_ENTRY(timer_from_check)
+TEST_ENTRY(timer_is_closing)
+TEST_ENTRY(timer_null_callback)
+TEST_ENTRY(timer_early_check)
+TASK_LIST_END
+
+@interface LibuvTimerTests : XCTestCase
+
+@end
+
+@implementation LibuvTimerTests
+
+- (void)testTimerAll {
+  for (task_entry_t* task = TASKS; task->main; task++) {
```

Comment:
we will need a way to obtain the test result signals from these test by calling into XCTest's assertion methods (https://developer.apple.com/documentation/xctest?language=objc).  Currently if we just run these it will always be reported as success (unless it crashed), thanks !

Reply:
The ASSERT works. If a test case fails at any assert, the whole test will crash. It'll be reported as success, only when all assert in all test cases passed.